engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java

```diff
@@ -147,7 +147,8 @@ public Query getLuceneQuery() {
 			DocumentExtractor extractor = buildDocumentExtractor( searcher, queryHits, first, max );
 			for ( int index = first; index <= max; index++ ) {
 				infos.add( extractor.extract( index ) );
-				if ( index % 10 == 0 ) {
+				//Check for timeout each 16 elements:
+				if ( (index & 0x000F) == 0 ) {
```

Comment:
same question as I had myself a couple of years ago :) no it seems JIT doesn't do that specific kind of optimisations. "%" is one of the operations we should always avoid according to our friends on OpenJDK.